docs/geps/gep-05.rst

```diff
@@ -0,0 +1,187 @@
++------------+------------------------------------------------------------------+
+| Resolution | <url> (required for Accepted or Rejected or Withdrawn)           |
++------------+------------------------------------------------------------------+
+
+
+Abstract
+--------
+
+This GEP describes the implementation of optional rounding of variables in GETTSIM.
+
+
+Motivation and Scope
+--------------------
+
+For several taxes and transfers, German law specifies that these be rounded in specific
+ways. This leads to different use cases for GETTSIM.
+
+1. Some applications require the exact, rounded, amount as specified in the law. This
+   is also helpful for creating test cases.
+2. Other applications require smooth functions and the non-rounding error does not
+   matter much.
+
+This document describes how we go about supporting both use cases.
+
+
+Implementation
+--------------
+
+GETTSIM allows for optional rounding of functions' results. Rounding parameters are
+specified in the ``.yaml``-files. The following goes through the details using an
+example from the basic pension allowance (Grundrente).
+
+The law on the public pension insurance specifies that the maximum possible
+Grundrentenzuschlag ``höchstwert_grundr_zuschlag_m`` be rounded to the nearest
+fourth decimal point (§76g SGB VI: Zuschlag an Entgeltpunkten für langjährige
+Versicherung). The example below contains GETTSIM's encoding of this fact.
+
+The snippet is taken from ``ges_rentenv.yaml``, which contains the following
+code:
+
+.. code-block:: yaml
+
+    rounding:
+      höchstwert_grundr_zuschlag_m:
+        2020-01-01:
+          base: 0.0001
+          direction: nearest
+          reference: §76g SGB VI Abs. 4 Nr. 4
+
+The specification of the rounding parameters starts with the key ``rounding`` at
+the outermost level of indentation. The keys are names of functions.
+
+At the next level, the ``YYYY-MM-DD`` key(s) indicate when rounding was
+introduced and/or changed. This is done in in the same way as for other policy
+parameters, see in :ref:`gep-3`. Those ``YYYY-MM-DD`` key(s) are
+associated with a dictionary containing the following elements:
+
+- The parameter ``base`` determines the base to which the variables is rounded.
+  It has to be a floating point number.
+- The parameter ``direction`` has to be one of ``up``, ``down``, or ``nearest``.
+- The ``reference`` must contain the reference to the law, which specifies the
+  rounding.
+
+In the same way as other policy parameters, the rounding parameters become part
+of the dictionary ``policy_params``.
+
+A function to be rounded must be decorated with ``add_rounding_spec``. This decorator
+indicates that the output should potentially be rounded. ``add_rounding_spec`` takes
+one required argument: ``params_key`` points to the key of the policy parameters
+dictionary containing the rounding parameters relating to the function that is
+decorated. In the above example, the rounding specification for
```

Comment:
Just for clarification: Is the 'key of the policy parameters dictionary' identical to the name of the `yaml` file?